### PR TITLE
Recording Added to Power Source

### LINKED ISSUE #1602: No recording functionality in Power Source
**Actual Behaviour**

No recording functionality in Power source

**Expected Behaviour**

Recording functionality should be added to power source

**Steps to reproduce it**

Open Power source

**LogCat for the issue**

Not Applicable

**Screenshots of the issue**

toast_issue

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by ho-dor**
Fixes #1602 

**Changes**: Recording function added to Power Source similar to Compass

**Screenshot/s for the changes**: 
power

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[Uploading app-debug_1.zip…]()

### CODE REVIEW

**Comment by ho-dor** [comment]
@CloudyPadmal @cweitat the single error in Codacy really should not be there because if I solve it and make the variable private , it will cause the app to crash as BindView can't be static or private . Can you help me understand how to prevent this Codacy error as all the BindView used in the app have package-private level . Thanks.

**Comment by cweitat** [comment]
Not all codacy needs to be fixed.

**Comment by ho-dor** [comment]
@CloudyPadmal @cweitat  can you please review this . This has been pending for sometime now , it is tagged high priority . Thank You.

File: app/src/main/java/io/pslab/models/PowerSourceData.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    public String getPcs() {
+        return pcs;
+    }
+
+    public void setPcs(String pcs) {
+        this.pcs = pcs;
+    }
+
```
Comment (abhinavraj23) [inline app/src/main/java/io/pslab/models/PowerSourceData.java]: Please remove these extra spaces

File: app/src/main/java/io/pslab/activity/PowerSourceActivity.java [hunk 2]
```diff
@@ -103,7 +116,7 @@
     ImageView bottomSheetSchematic;
     @BindView(R.id.custom_dialog_desc)
     TextView bottomSheetDesc;
-
+    private PowerSourceData powerSourceData = new PowerSourceData();
```
Comment (abhinavraj23) [inline app/src/main/java/io/pslab/activity/PowerSourceActivity.java]: @ho-dor Why are you creating the object directly here, shouldn't it be more appropriate if it is declared in onCreate method

**Comment by neel1998** [comment]
@abhinavraj23  I also agree with you. I don't see any need of adding recording functionality to Power Source

**Comment by ho-dor** [comment]
@mariobehling tagged it as high priority i.e. why I implemented this.

**Comment by neel1998** [comment]
@ho-dor  i have seen the tag. Still i don't clearly see the need of it. @mariobehling  kindly send some light

**Comment by ho-dor** [comment]
@Avjeet please review this

**Comment by ho-dor** [comment]
@cweitat please review this

**Pull request merged**